toolchain/check/lexical_lookup.h

```diff
@@ -44,7 +52,26 @@ class LexicalLookup {
     return lookup_[index];
   }
 
+  // Temporarily remove the top lookup result for `name_id` from scope.
+  auto Suspend(SemIR::NameId name_id) -> SuspendedResult {
+    auto index = GetLookupIndex(name_id);
+    auto& results = lookup_[index];
+    CARBON_CHECK(!results.empty())
+        << "Suspending a nonexistent result for " << name_id << ".";
+    return {index, results.pop_back_val().inst_id};
+  }
+
+  // Restore a previously-suspended lookup result.
+  auto Restore(SuspendedResult sus, ScopeIndex index) -> void {
+    lookup_[sus.index].push_back({sus.inst_id, index});
+  }
+
  private:
+  // Get the index at which the specified name is stored in `lookup_`.
+  auto GetLookupIndex(SemIR::NameId name_id) -> size_t {
+    return name_id.index + SemIR::NameId::NonIndexValueCount;
```

Comment:
```suggestion
  auto GetLookupIndex(SemIR::NameId name_id) -> int32_t {
    return name_id.index + SemIR::NameId::NonIndexValueCount;
```

Isn't name_id.index an int32_t? Why does this use size_t?

Reply:
It's converting a `NameId` to an array index, which (for `SmallVector`) is expected to be a `size_t`. (The internal representation of `NameId` doesn't seem important here.)

Reply:
Most of my thought is that it seems a little inconsistent with surrounding code. But also, you do:

```
auto index = GetLookupIndex(name_id);
auto& results = lookup_[index];
...
return {static_cast<uint32_t>(index), results.pop_back_val().inst_id};
```

So yes it's used as an array index, but because it comes from a NameId required to fit inside a uint32_t. Additionally, it's natively a int32_t. So it's being upsized and downsized, but it's practically restricted to be a valid int32 (arguably, max values might overflow, but that feels really edge-casey).

Anyways, I'm not insisting here, but maybe it's worth thinking about that you have something that must be a valid int32_t, is converted to a size_t, then to a uint32_t. int32_t stores compactly and would be a consistent type.